pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/readers/CompactedPinotSegmentRecordReader.java

```diff
@@ -36,19 +37,25 @@
 public class CompactedPinotSegmentRecordReader implements RecordReader {
   private final PinotSegmentRecordReader _pinotSegmentRecordReader;
   private final RoaringBitmap _validDocIdsBitmap;
-
+  private final String _deleteRecordColumn;
+  // Reusable generic row to store the next row to return
+  private final GenericRow _nextRow = new GenericRow();
   // Valid doc ids iterator
   private PeekableIntIterator _validDocIdsIterator;
-  // Reusable generic row to store the next row to return
-  private GenericRow _nextRow = new GenericRow();
   // Flag to mark whether we need to fetch another row
   private boolean _nextRowReturned = true;
 
   public CompactedPinotSegmentRecordReader(File indexDir, RoaringBitmap validDocIds) {
+    this(indexDir, validDocIds, null);
+  }
+
+  public CompactedPinotSegmentRecordReader(File indexDir, RoaringBitmap validDocIds,
+      @Nullable String deleteRecordColumn) {
```

Comment:
@klsince you can make it an Upsert Compaction task-level config as well if you want to use this flag flexibly in your cluster. By default, we can disable it.

Reply:
As it’s not safe to skip deleteRecord, so I would rather avoid it for less confusion.